features/atcryptoauth/atca_app.cpp

```diff
@@ -0,0 +1,304 @@
+ *  interface with following options:
+ *  a) Commission ATCAECC508A with:
+ *          - slots 0-7 as privates keys.
+ *          - slots 9-14 as public keys.
+ *          - slot 8 as certificate store.
+ *  b) Dump configuration registers.
+ *  a) Generate private key in given slot.
+ *  a) Run an SSL client example.
+ *
+ *  @return Implements a serial input loop and never returns.
+ */
+int main()
+{
+    size_t magic_pattern_idx = 0;
+    mbed::RawSerial * serial = new mbed::RawSerial(USBTX, USBRX, MBED_CONF_PLATFORM_STDIO_BAUD_RATE);
+    ATCA * atca = ATCA::GetInstance();
+    ATCAError err = ATCA_ERR_NO_ERROR;
+
+    printf("\r\n*****************************************\r\n");
+    printf(    "*          MBED ATCA ECC App           **\r\n");
+    printf(    "*****************************************\r\n");
+
+    while( 1 )
+    {
+        /* Print menu */
+        if ( magic_pattern_idx == 0 )
+        {
+            printf("\r\nOptions:\r\n");
+            printf("a) Commission ATCAECC508A.\r\n");
+            printf("b) Dump config.\r\n");
+            printf("c) Generate Private key.\r\n");
+            printf("d) Run SSL client example.\r\n");
+            printf("\r\nEnter choice: ");
+            fflush(stdout);
+        }
+        char c = serial->getc();
+        serial->putc( c ); /* echo received input */
+        switch( c )
+        {
+            case 'a':
+                {
+                    printf("\r\nCommissioning...\r\n");
+                    if ( ( err = atca->Commission( )) != ATCA_ERR_NO_ERROR )
+                    {
+                        printf ("Failed to commission! %04X\r\n", err);
+                        break;
+                    }
+                    printf ("\r\nConfiguration successful. Canfig zone can be locked now.\r\n"
+                            "\r\nWarning!!! Locking the config is irreversible!\r\n"
+                            "\r\nPress 'y' for locking or any key to skip: ");
+                    fflush( stdout );
+                    if ( serial->getc() == 'y' )
+                    {
+                        err = atca->LockConfig();
+                        if (err != ATCA_ERR_NO_ERROR)
+                        {
+                            printf ("Failed to lock config! %04X\r\n", err);
+                            break;
+                        }
+                    }
+                }
+                break;
+            case 'b':
+                {
+                    printf ("\r\nATCAECC508A configuration dump:\r\n");
+                    atca->DumpDeviceConfig();
+                }
+                break;
+            case 'c':
+                {
+                    if ( atca->GenPrivKey( ATCA_KEY_ID_0 ) != ATCA_ERR_NO_ERROR )
+                        printf("Failed to generated key at slot 0. error %04X\r\n", err);
+                    else
+                        printf("\r\nThis sample generates only one private key\r\n"
```

Comment:
This message is misleading: it looks like general documentation but actually indicates that the private key was generated.